Implementation/Services/index.md

```diff
@@ -369,51 +437,80 @@ namespace Umbraco8.Controllers
 {
     public class BlogPostController : Umbraco.Web.Mvc.RenderMvcController
     {         
-        private readonly ISiteHelperService _siteHelperService;
-        private readonly ILogger _logger;
+        private readonly ISiteService _siteService;
 
-        public BlogPostController(ILogger logger, ISiteHelperService siteHelperService)
+        public BlogPostController(ISiteService siteService)
         {
-            _logger = logger;
-            _siteHelperService = siteHelperService; 
+            _siteService = siteService ?? throw new ArgumentNullException(nameof(siteService)); 
         }
 
         public override ActionResult Index(ContentModel model)
         {
-            var newsSection = _siteHelperService.GetNewsSection();
+            var newsSection = _siteService.GetNewsSection();           
             var blogPostViewModel = new BlogPostViewModel(model);
-            blogPostViewModel.NewsSection = newsSection;
+            blogPostViewModel.HasNewsSection = false;
+            if (newsSection!=null){
+                blogPostViewModel.HasNewsSection = true;
+                blogPostViewModel.NewsSection = newsSection;
+            }
             //etc          
             // Do other stuff here!, then return the custom viewmodel to the template view.
             return CurrentTemplate(blogPostViewModel);
-
         }
     }
 }
 ```
+:::warning
+This isn't truly 'best practice' when using DI. This 'only works' in Umbraco because when a dependency for the base RenderMvcController isn't supplied via a constructor, Umbraco 'falls back' and uses the Service Locator pattern to inject the missing elements. This enables developers to choose to ignore DI, but if trying to following DI best practice, and to make the controller 'unit testable' - use the following example instead.
```

Comment:
should say "use the following example instead which supplies all constructor parameters for the base class."